tasks/queue-granules/index.js

```diff
@@ -20,6 +25,30 @@ async function fetchGranuleProvider(prefix, providerId) {
   return JSON.parse(body);
 }
 
+/**
+ * Group granules by collection and split into batches then split again on provider
+ *
+ * @param {Array<Object>} granules - list of input granules
+ * @param {number} [batchSize] - size of batch of granules to queue
+ * @returns {Array<Object>} list of lists of granules: each list contains granules which belong
+ *                          to the same collection, and each list's max length is set by batchSize
+ */
+function groupAndBatchGranules(granules, batchSize) {
+  const filteredBatchSize = isNumber(batchSize) ? batchSize : 1;
```

Comment:
```suggestion
function groupAndBatchGranules(granules, batchSize = 1) {
```
Could we not just set the default on the parameter in the function?

Reply:
@Jkovarik knows the details better but I think we were running into trouble when not doing an explicit check.

Reply:
No, we can't.    The reason is if we configure the preferredBatchSize value in the workflow configuration, but *some* rule invocations don't have it in the meta, it winds up passing `null` (None in python when it evals the object) -- this results in batch size *being* defined as null.  

I wrote a unit to validate this behavior, you can try it out if you'd like to see what happens, the unit will fail.     Hilariously it winds up *queuing nothing* if we don't do this eval.